### PR TITLE
Kosta/promo code

### PR DESCRIPTION
**Pull request by kostamas**

### CODE REVIEW

File: src/components/promoCode/services/promoCodeSrv.service.js [hunk 1]
```diff
@@ -26,7 +33,7 @@
                 promoCodeStatusText[INVALID] = INVALID;
 
                 promoCodeSrv.checkPromoCode = function (promoCode, appContext) {
-                    var firebaseAppScopeName =  backendData[appContext].firebaseAppScopeName;
+                    var firebaseAppScopeName = backendData[appContext].firebaseAppScopeName;
```
Comment (Ariel-W) [inline src/components/promoCode/services/promoCodeSrv.service.js]: You set:
appContext = ENV.firebaseAppScopeName

So where is backendData[appContext].firebaseAppScopeName initialized?
And why do you have both:
backendData[appContext].firebaseAppScopeName
AND
appContext = ENV.firebaseAppScopeName

File: dist/main.js [hunk 2]
```diff
@@ -13022,7 +13062,7 @@ angular.module('znk.infra-web-app.onBoarding').run(['$templateCache', function (
                 promoCodeStatusText[INVALID] = INVALID;
 
                 promoCodeSrv.checkPromoCode = function (promoCode, appContext) {
-                    var firebaseAppScopeName =  backendData[appContext].firebaseAppScopeName;
+                    var firebaseAppScopeName = backendData[appContext].firebaseAppScopeName;
```
Comment (assafshp) [inline dist/main.js]: in one place you use firebaseAppScopeName and other appName - > has to be consistent

File: src/components/promoCode/services/promoCodeSrv.service.js [hunk 3]
```diff
@@ -2,8 +2,15 @@
     'use strict';
 
     angular.module('znk.infra-web-app.promoCode').provider('PromoCodeSrv',
-        function () {
+        function (ENV) {
+
             var backendData = {};
+            var appContext = ENV.firebaseAppScopeName;
+
+            backendData[appContext] = {  //default data
```
Comment (assafshp) [inline src/components/promoCode/services/promoCodeSrv.service.js]: I don't understand why you need this obj map ? in infra-web-app you automaticlly get ENV params based on your existing app context, no need to get the app....
or I'm missing something

File: src/components/promoCode/run/promocode.run.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        .run(function ($location, PromoCodeSrv, AuthService, PopUpSrv, $filter, ENV) {
+        'ngInject';
+        var authData = AuthService.getAuth();
+        var translate = $filter('translate');
+
+
+        if (authData && authData.uid) {
+            var search = $location.search();
+            var promoCodeId = search.pcid;
+            var appContext = ENV.firebaseAppScopeName;
```
Comment (assafshp) [inline src/components/promoCode/run/promocode.run.js]: ENV.firebaseAppScopeName or ENV.appName ?

File: src/components/promoCode/services/promoCodeSrv.service.js (hunk 1 above)
Comment (kostamas) [inline src/components/promoCode/services/promoCodeSrv.service.js]: there are 2 places using promo code srv, and the use is different: 
1. in website - i need the all apps env and there i initializing the backendData[appContext].firebaseAppScopeName (in config phase). thats the reason i need the appContext map.
2. in app (student/dashboard) - i use the default backendData and i need only the appName (it is the student fb path).

File: src/components/promoCode/services/promoCodeSrv.service.js (hunk 3 above)
Comment (kostamas) [inline src/components/promoCode/services/promoCodeSrv.service.js]: there are 2 places using promo code srv, and the use is different:

1.in website - i need the all apps env and there i initializing the backendData[appContext].firebaseAppScopeName (in config phase). thats the reason i need the appContext map.
2.in app (student/dashboard) - i use the default backendData and i need only the appName (it is the student fb path).

**Pull request merged**